crates/ruff_linter/src/rules/ruff/rules/unnecessary_regular_expression.rs

```diff
@@ -0,0 +1,243 @@
+            ReFuncKind::Match => self.method_expr("startswith", vec![self.pattern.clone()]),
+            // pattern in string
+            ReFuncKind::Search => self.compare_expr(CmpOp::In),
+            // string == pattern
+            ReFuncKind::Fullmatch => self.compare_expr(CmpOp::Eq),
+            // string.split(pattern)
+            ReFuncKind::Split => self.method_expr("split", vec![self.pattern.clone()]),
+        }
+    }
+
+    /// Return a new compare expr of the form `self.pattern op self.string`
+    fn compare_expr(&self, op: CmpOp) -> Expr {
+        Expr::Compare(ExprCompare {
+            left: Box::new(self.pattern.clone()),
+            ops: Box::new([op]),
+            comparators: Box::new([self.string.clone()]),
+            range: TextRange::default(),
+        })
+    }
+
+    /// Return a new method call expression on `self.string` with `args` like
+    /// `self.string.method(args...)`
+    fn method_expr(&self, method: &str, args: Vec<Expr>) -> Expr {
+        let method = Expr::Attribute(ExprAttribute {
+            value: Box::new(self.string.clone()),
+            attr: Identifier::new(method, TextRange::default()),
+            ctx: ExprContext::Load,
+            range: TextRange::default(),
+        });
+        Expr::Call(ExprCall {
+            func: Box::new(method),
+            arguments: Arguments {
+                args: args.into_boxed_slice(),
+                keywords: Box::new([]),
+                range: TextRange::default(),
+            },
+            range: TextRange::default(),
+        })
+    }
+}
+
+/// RUF055
+pub(crate) fn unnecessary_regular_expression(checker: &mut Checker, call: &ExprCall) {
+    // adapted from unraw_re_pattern
+    let semantic = checker.semantic();
+
+    if !semantic.seen_module(Modules::RE) {
+        return;
+    }
+
+    let Some(qualified_name) = semantic.resolve_qualified_name(call.func.as_ref()) else {
+        return;
+    };
+
+    let ["re", func] = qualified_name.segments() else {
+        return;
+    };
+
+    // skip calls with more than `pattern` and `string` arguments (and `repl`
+    // for `sub`)
+    let Some(re_func) = ReFunc::from_call_expr(checker, call, func) else {
+        return;
+    };
+
+    let Some(pat) = call.arguments.find_argument("pattern", 0) else {
+        // this should be unreachable given the checks above, so it might be
+        // safe to unwrap here instead
+        return;
+    };
+
+    // For now, restrict this rule to string literals
+    let Some(string_lit) = pat.as_string_literal_expr() else {
+        return;
+    };
+
+    // For now, reject any regex metacharacters. Compare to the complete list
+    // from https://docs.python.org/3/howto/regex.html#matching-characters
+    let has_metacharacters = string_lit.value.chars().any(|c| {
+        matches!(
+            c,
+            '.' | '^' | '$' | '*' | '+' | '?' | '{' | '}' | '[' | ']' | '\\' | '|' | '(' | ')'
+        )
+    });
```

Comment:
I like how you intentionally excluded meta-characters. So consider this an extension, and I think it's totally fine to do this as a follow-up pr (or not at all). 

It would be nice if the rule only skips replacement for characters that are different between regex expressions and regular strings. For example, `\n` matches `\n` in a regex and a string.